base_layer/core/src/base_node/sync/header_sync/synchronizer.rs

```diff
@@ -402,11 +402,11 @@ impl<'a, B: BlockchainBackend + 'static> HeaderSynchronizer<'a, B> {
             .map_err(BlockHeaderSyncError::ReceivedInvalidHeader)?;
         let num_new_headers = headers.len();
 
-        // We can trust that the header associated with this hash exists because block_hashes is data this node
+        // NOTE: We can trust that the header associated with this hash exists because block_hashes is data this node
```

Comment:
This sentence does not read right

Reply:
Could be better - updated it, basically it's saying that unwrap is safe because we know/control what's in block_hashes